### ISSUE TITLE
Internal Server Error

### ISSUE THREAD

**Issue by ulli-steiner** [ORCID-EU-Labs/orcid-feed#14]
The command 

http://feed.labs.orcid-eu.org/0000-0001-5936-339X.json

produces an internal server error.  The ID is however legitimate, i.e.

http://orcid.org/0000-0001-5936-339X

produces the expected result.

Ulli

**Comment by mfenner** [comment]
Thank you, I will look into that. I expect that you have some unusual formatting in your ORCID profile.

**Comment by ulli-steiner** [comment]
Martin,

found the culprit.  It is down to the fact that adding elements “by hand” leads to a fault in the BibTex code.  It is a consequence that ORCID presently does not have a good interface for adding records and none to modify them.

Thanks

Ulli

**Comment by mfenner** [comment]
Do you still want me to do anything? I tried to make my code as flexible as possible to handle the various input formats. But I admit that I didn't test manual input, because I'm personally not a big fan of manual input of bilbliographic data.

**Comment by ulli-steiner** [comment]
Not really.  json and bib work now. XML, for some reason gives a bibtex error, but I don;t care.  

I am planning to us my public ORCID record to create a publication list my web page (which I currently do in a very unsatisfactory way using my own database).  For this, I have to be able to add ORCID records that are not collected by the usual aggregators.  I have developed my own  javascript that uses the data from http://pub.orcid.org/0000-0001-5936-339X/orcid-works. I am planning to switch to you feed, since it is more elegant than my home-cooked solution.

If you have come good contacts within the ORCID team, maybe you can talk them into improving their interface.  My main grips are that duplicate records have to be removed “by hand”, the manual addition of records is clunky, and there is no way to edit these once they are added.  

I have send in my suggestions a few times already:  (1) Mark manual records as such, (2) Allow to edit manual records only, (3) identify duplicate records by their doi, ISBN, ISSN numbers.  I hope anyone listens.

Thanks for replying,

Ulli

**Comment by mfenner** [comment]
Ulli, thanks for the feedback, I will forward your comments to the ORCID support team. If you haven't done so already, you can also leave your comments at http://support.orcid.org/.

I've built the ORCID Feed application partly so that I can semi-automatically display my publications on my blog: http://blog.martinfenner.org/about.html. I use a combination of Javascript and Ruby.

**Comment by gthorisson** [comment]
Hi guys. There’s also some Javascript here, used with the HTML-output of the orcid-feed service, which was put together at the ODIN hackathon back in October (the "Embedding all my ORCID-claimed works” project):

http://odin-project.eu/project-outputs/first-codesprint-oct-2013/

needs a bit of polish to be super-easy to drop into a page, but the main bits are there if you want to experiment.

The JS pulls in formatted, hyperlinked citations as HTML from Martin’s tool and inserts into the page. Here’s some example HTML-output if you want to use this in your own JS, Ulli:

http://feed.labs.orcid-eu.org/0000-0003-1419-2405.html

```
    Mummi
```

**Comment by ulli-steiner** [comment]
Mummi,

thank you for this.  For everyone’s info:  I used manual input into my ORCID publication list, using the BibTex output from the publisher’s web page (publishers that do not provide a direct ORCID link).  Unfortunately, the automatically generated BibTex is very often rubbish.  In these particular cases, the records had empty cite-keys, which upset the feed.  

Ulli

**Comment by mfenner** [comment]
Empty citekeys is something I can fix in ORCID Feed. Mummi and I had a related issue once where citekeys for data started with numbers, confusing the ORCID service on import.

**Comment by ulli-steiner** [comment]
One more thing:

the json feed does well in capturing all relevant fields from records that were uploaded from Scopus.  It does less well when the records stem from ResearcherID, where none of the digital identifiers (DOI, ISBN,..) make it into the feed.

On another matter.  Before putting my project online:  what is the right way to acknowledge ORCID, and/or the feed?

Ulli

**Comment by lpaglione** [comment]
Hi -- Sorry for the delay in getting back to you. Thanks for your question
about acknowledgements.

For acknowledgement we don't have anything specific for this particular
case, though for acknowledgements in code, see:

http://orcid.org/open-source-license

For acknowledgements otherwise, we suggest just a thank you to ORCID. Some
guidelines on the use of our name can be seen here:

http://orcid.org/trademark-and-id-display-guidelines

Related to your earlier question about duplicates, since our primary goal
is to provide a linking mechanism, we actually consider information from
different sources (scopus/reseracherID/etc) about, say, an article, to be
different metadata, not a duplicate. We have been working with a working
group on strategies for grouping this information to make it clearer to
those consuming data directly from an ORCID record, and to allow ORCID iD
holders to specify a primary version of the metadata to be used in
applications such as the one you have developed. This functionality is just
in the design stages, though we will include announcement about it in our
blog and mailing lists when it's ready. If you haven't already, you may be
interested in subscribing to one or both of the following lists:
- ORCID Blog - http://orcid.org/newsletter/subscriptions
- ORCID API Users forum -  https://groups.google.com/d/forum/orcid-api-users

Thanks!

Best, L

---

Laura Paglione
Technical Director, ORCID
L.Paglione@orcid.org
+1-301-500-2139
http://orcid.org/0000-0003-3188-6273 http://orcid.org/0000-0003-3188-6273

**Comment by ulli-steiner** [comment]
Dear all,

There seems to be an occasional problem with DOI parsing:  The DOI shows up in ORCID, but not in "feed.labs.orcid-eu.org”.

Examples:

ORCID:   http://orcid.org/0000-0001-5936-339X

The following records have DOI entries:

Bio-inspired hierarchical polymer fiber-carbon nanotube adhesives 2014

Labyrinth-Induced Faceted Electrochemical Growth 2014

Room-temperature development of thin film composite reverse osmosis membranes from cellulose acetate with antibacterial properties: Journal of Membrane Science 2014

Pointillist structural color in Pollia fruit: Proceedings of the National Academy of Sciences of the United States of America 2012

Patterning self-assembled monolayers on oxide surfaces using a lift-off technique: Advanced Materials 1999

When invoking the corresponding feed:  http://feed.labs.orcid-eu.org/0000-0001-5936-339X.json

The DOI entries are missing.  

Any advice?

Happy Easter!

Ulli

**Comment by naa** [comment]
Dear @mfenner ,
I'm using orcid-feed to implement automatic publication lists on the personal webpages of people in my department. Unfortunately, some people have entered older papers by hand and I have the problem described by @ulli-steiner , e.g. http://feed.labs.orcid-eu.org/0000-0002-2561-4610.js
Is it possible to get at least all other publications for such a profile?
